### ISSUE TITLE
Use BUILD_DSP_G register to determine which Base[#] are present

### ISSUE THREAD

**Issue by jmdewart** [slaclab/pysmurf#48]
FW includes an 8 bit register BUILD_DSP_G.  0xFF means Base[0...7] are built, 0xF Base[0...3], etc.  Example:

cryo@smurf-srv11:/$ caget -lx smurf_server_s4:AMCc:FpgaTopLevel:AppTop:AppCore:BUILD_DSP_G
smurf_server_s4:AMCc:FpgaTopLevel:AppTop:AppCore:BUILD_DSP_G 15

15 -> 0xFF Base[0...7] present

**Comment by swh76** [comment]
Currently which bands are being used is determined from which "band_" blocks are present in the pysmurf config.  Maybe we should switch to using this, instead.

**Comment by swh76** [comment]
Added get_build_dsp_g function and which_bands function to return the BUILD_DSP_G register and to return a list of the bands encoded by BUILD_DSP_G in commit https://github.com/slaclab/pysmurf/commit/f6e94845eaf2d2ab2a6c30ed30395f8b42a0db52 of the dspv3 branch.  Not yet used.

**Comment by ruck314** [comment]
The two helpers requested in this issue have been on `main` since 2019:

- `get_build_dsp_g()` — `python/pysmurf/client/command/smurf_command.py:1763-1773`
- `which_bands()` — `python/pysmurf/client/util/smurf_util.py:2419-2429`

`which_bands()` is consumed in two production sites today (`smurf_util.py:3971` to map a tone frequency to a band, and `smurf_util.py:4219` to derive bays for AMC temperature monitoring), so the original ask — exposing `BUILD_DSP_G` from pysmurf — is implemented.

While reviewing for closure I noticed a latent bit-ordering bug in `which_bands()`:

```python
bands=[b for b,x in enumerate(bin(build_dsp_g)[2:]) if x=='1']
```

`bin(x)[2:]` is MSB-first with no zero padding, so the `enumerate` index does not correspond to the bit position. It happens to be correct for masks that are contiguous from the LSB (`0x0F`, `0xFF`) — which covers every shipped firmware build I'm aware of and is why this went unnoticed — but it misreports any other mask:

| `build_dsp_g` | expected | current return |
|---|---|---|
| `0xF0` | `[4,5,6,7]` | `[0,1,2,3]` |
| `0x80` | `[7]` | `[0]` |
| `0x10` | `[4]` | `[0]` |

A one-line fix (iterate `range(8)` explicitly with shift/mask) is staged on branch `issues-48` and will follow as a PR.

The remaining open thread from @swh76's 2019 comment — replacing the config-driven `init.bands` list with `which_bands()` as the canonical source of truth — is a separate architectural change. It affects every site `cfg` file and the existing `bays`/`usable_bands` cross-check in `smurf_control.py:264-298`, and should be tracked under its own issue.

Recommend closing #48 once the bit-ordering fix lands.
